cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -435,6 +437,8 @@ protected void configure() {
       handlerBinder.addBinding().to(TetheringServerHandler.class);
       handlerBinder.addBinding().to(TetheringClientHandler.class);
       handlerBinder.addBinding().to(AppStateHandler.class);
+      handlerBinder.addBinding().to(CredentialProviderHttpHandler.class);
```

Comment:
should we add these handlers only if system workers are disabled?

Reply:
Removed system workers usage after discussing offline with @albertshau.